lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1298,8 +1298,17 @@ extension DomScreenExtension on DomScreen {
 class DomScreenOrientation extends DomEventTarget {}
 
 extension DomScreenOrientationExtension on DomScreenOrientation {
-  Future<dynamic> lock(String orientation) => js_util
-      .promiseToFuture(js_util.callMethod(this, 'lock', <String>[orientation]));
+  Future<dynamic> lock(String orientation) {
+    final Object jsResult = js_util.callMethod(this, 'lock', <String>[orientation]);
+    // `allowInterop` does not automatically convert Future to Promise. So when mocking
+    // this method in tests the returned Future doesn't need to be converted. To anyone
+    // reading this in the future, if you know of a better way, feel free to fix this.
+    // This special-case doesn't feel right.
+    if (jsResult is Future) {
+      return jsResult;
+    }
```

Comment:
This check seems fine to me, but I don't know if this is the right place for it. Maybe `js_util.promiseToFuture` should do this check internally? I'm not sure.

Or like you said, maybe `allowInterop` should convert futures to promises.

Reply:
Actually, I changed the test to use the `Promise` API directly, so there's no futures to convert. Looks much better now. The point about `allowInterop` limitation still stands though, although less relevant to this PR now.